### PR TITLE
ARROW-3225: [C++/Python] Pandas object conversion of ListType<DateType> and ListType<TimeType>

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: cpp/src/arrow/python/python_to_arrow.cc [hunk 1]
```diff
@@ -313,16 +313,61 @@ class Date64Converter : public TypedConverter<Date64Type, Date64Converter> {
...
+      // datetime.time stores microsecond resolution
+      switch (unit_) {
+        case TimeUnit::SECOND:
+          t = static_cast<int32_t>(PyTime_to_s(obj));
+          break;
+        case TimeUnit::MILLI:
+          t = static_cast<int32_t>(PyTime_to_ms(obj));
+          break;
+        default:
+          return Status::UnknownError("Invalid time unit");
```
Comment (wesm) [inline cpp/src/arrow/python/python_to_arrow.cc]: There should probably be "DCHECK(false)" here also

File: cpp/src/arrow/python/python_to_arrow.cc [hunk 2]
```diff
@@ -313,16 +313,61 @@ class Date64Converter : public TypedConverter<Date64Type, Date64Converter> {
...
-      return typed_builder_->Append(PyTime_to_us(obj));
+      switch (unit_) {
+        case TimeUnit::MICRO:
+          t = PyTime_to_us(obj);
+          break;
+        case TimeUnit::NANO:
+          t = PyTime_to_ns(obj);
+          break;
+        default:
+          return Status::UnknownError("Invalid time unit");
```
Comment (wesm) [inline cpp/src/arrow/python/python_to_arrow.cc]: same

**Review by wesm — APPROVED** [review]
+1, with small comment. let me know if you want to make any changes; either way is fine

**Comment by kszucs** [comment]
Ehh, just was about to add them :)
Do You want me to create a PR for that?

**Comment by wesm** [comment]
Ah, it's no big deal, but feel free to if you wish

**Pull request closed without merging**